### DISCUSSION TITLE
weather widget always shows as people app.

### DISCUSSION THREAD

**Discussion by chip33** [valinet/ExplorerPatcher#823] [Q&A]
Windows version latest stable :22000.469
EP 22000.469.42.7

The weather widget was working fine for me up to and including the [commit ](https://github.com/valinet/ExplorerPatcher/commit/74204c7e84dc63d28fed0a6022225bcec3b60cdb) where dark support was added.
After that I had the issue where explorer would crash permanently on EP installation so I stayed on a version pre weather widget until [22000.469.42.6](https://github.com/valinet/ExplorerPatcher/releases/tag/22000.469.42.6_16ccb3f)

Since then I only see the people widget when I enable weather widget.
I have reinstalled Evergreen WebView2 standalone component but the issue still persists through reboots.

I don't see anything unusual in the console output. and have tried uninstalling ep clearing all settings and reinstalling.

**Comment by valinet** [comment]
Well, idk, it has to be debugged. Maybe there's an oversight on my side, idk. Do you see WebView2 processes when `explorer` runs? Try to enable [full crash dumps](https://docs.microsoft.com/en-us/windows/win32/wer/collecting-user-mode-dumps), collect one and look on that or maybe upload it here... it has to be debugged, I can't think hat it could be, the architecture is pretty new, maybe there are still corner cases which have to be ironed out.

Thanks.

**Reply by chip33** [reply]
Thanks. I have 5 "Microsoft Edge Webview2" processes in Task Scheduler , they are there whether EP is installed or not.
Will enable full crash dumps and collect one.

**Reply by valinet** [reply]
Use Process Hacker 2 or Process Monitor which can show a tree for the processes, and in there identify to which application those WebView2 processes belong. If they're under `dllhost.exe ... /ProcessId:{A6EA...`, than that's the weather widget. When you enable the console, that should display a console window as well, if not, then maybe it crashed. If it doesn't crash, attaching with a debugger and stepping through might show something relevant.

**Reply by chip33** [reply]
Update! after rolling back to 22000.469.42.0. I got an EP update notification for latest pre-release 469.42.8 and because EP updated itself I got the console screen flash up and Weather is back.

It seems manually updating EP from the (pre-) releases page on github may have been part of what stopped the weather widget working. Somehow that was different than when ep updates itself?

--------

This is what I wrote before it got fixed.

I think the weather widget crashes as all the WebView2 processes are under widgets.exe at all times and only one console window opens.

I rolled back through ep versions:  the "permanent" explorer crash begins for me with ep [22000.469.42.0](https://github.com/valinet/ExplorerPatcher/releases/tag/22000.469.42.0_5d44661)

and the weather widget now always displays as people even with 22000.469.42.0.
I remember now it did crash then but i restarted explorer and after seeing the people icon briefly i got the weather widget and then the dark mode update.

After that the crashes became so bad I waited until [22000.469.42.6](https://github.com/valinet/ExplorerPatcher/commit/16ccb3f839db038e9391e423130eb00bca40d315)
before I updated and have not been able to see the weather since :)
Here are some windbg outputs from 2 explorer crashes with 22000.469.42.0 installs.


[windbg_ep_22000.469.42.0.txt](https://github.com/valinet/ExplorerPatcher/files/8027202/windbg_ep_22000.469.42.0.txt)

**Reply by valinet** [reply]
Hi. Yeah, it may have been a problem with it setting up, some versions had a bug in the installer, these latest ones should correct that.

The updates delivered by GitHub in the “Releases” section are the same as the ones fetched via the update mechanism in the app.

Anyway, I am glad that you got it sorted out.

**Reply by chip33** [reply]
Thanks, I just wondered , does the windbg_ep_22000.469.42.0.txt file tell you much that is useful?

Yes, I had a feeling I wasn't getting updates correctly but wasn't sure as I was using an older release. I'm glad to have it working again.

**Reply by valinet** [reply]
Well, it actually does. It would have been better if WinDbg had access to symbols for ExplorerPatcher, but in this instance it is okay nevertheless, as there are symbols downloaded for the rest of the DLLs, and with those I can see in which hooked function the crash is happening.

First of all, EP's version that you had was 42.0, you can easily tell that from WInDbg's output. That's pretty buggy, in regards to the weather widget, as I was not protecting quite a few variables against race conditions, and that was precisely what was happening. I mean, I was aiming to get something done an then fix it as we moved along. I was getting a lot of those crashes as well on that version, so I kind of knew. These should be (are) solved in the latest pre-releases.

Maybe the installer was failing to update the files (that was buggy as well until the version I released last Sunday) and that is why you were still getting an old version despite seeming like it succeeded. Yeah, idk, that's the thing, it's only me that tests these, I don't have much time sometimes, other times I might miss things, but since I daily drive it as well, it eventually should stabilize, as I go on and fix the bugs, since they bother me as well, if they're there.

**Comment by gfmoore** [comment]
I am on win 10 taskbar and now the ridiculous people icon is showing instead of weather icon. I thought weather icon was showing on win 10 yesterday, but can't be sure. No option to remove it that I can see on taskbar settings.

**Comment by gfmoore** [comment]
oops just done an update and its back :), the weather that is.
